server/src/watchdog.js

```diff
@@ -107,14 +107,53 @@ exports.submit = function(shot) {
   const imageId = imageUrl.split("/").pop();
   const form = new FormData();
 
-  Shot.getRawBytesForClip(
-    imageId
-  ).then((obj) => {
-    if (obj === null) {
-      const err = new Error("Unable to fetch image for Watchdog submission.");
-      err.imageId = String(imageId);
-      throw err;
-    }
+  // When this is called the shot has been successfully saved to the database,
+  // but the image data is not necessarily on S3 as the save to S3 is not
+  // awaited. Awaiting the S3 upload will affect the responsiveness of saving a
+  // shot with the addon.
+  //
+  // We'll give the S3 upload a head start and try a few times to fetch the
+  // image data. This is called for every shot saved, so let's keep it simple
+  // and minimal.
+
+  const getImageRawBytes = function() {
+    const MAX_ATTEMPTS = 3;
+    const DELAY = 1000; // milliseconds
+    let attemptCount = 1;
+
+    const delay = function(waitInMs) {
+      return new Promise((resolve) => {
+        setTimeout(() => resolve(), waitInMs);
+      });
+    };
+    const getRawBytesPromise = function(timeToWait) {
```

Comment:
Probably couldn't hurt to move `getRawBytesPromise` and the `delay` function out to top level, to avoid redeclaring them over and over.

Reply:
hmmm I did that to minimize the scope of the functions.  I think it wouldn't hurt anything?  I'll move them up.